Detectors/FIT/FT0/simulation/src/Digits2Raw.cxx

```diff
@@ -252,3 +253,11 @@ EventHeader Digits2Raw::makeGBTHeader(int link, o2::InteractionRecord const& mIn
   }
   return mEventHeader;
 }
+
+//_____________________________________________________________________________________
+int Digits2Raw::carryOverMethod(const header::RDHAny* rdh, const gsl::span<char> data,
+                                const char* ptr, int maxSize, int splitID,
+                                std::vector<char>& trailer, std::vector<char>& header) const
+{
+  return data.size() > maxSize ? 0 : data.size();
```

Comment:
The ``data`` is the full original payload you are providing to RawWriter, the ``maxSize`` is the min of the number of bytes left on the CRU page and still unwritten payload. The method is called only when the payload does not fit
into the remaining CRU page, i.e. if the method is called, then by definition ``data.size() > maxSize``.
Therefore, such implementation will always return 0, forcing to close current CRU page and write the rest of data on new page. In case your payload exceeds the 8 KB, this will lead to infinite loop.

Instead, you have to decrease the maxSize in such a way that you avoid to split the data in the region which should not be split (e.g. isolated header or even its part). Could you tell me what are the constraints on splitting your data, I will suggest what to do?

Reply:
FT0 data size never can be greater 8K. So we don't need to split data,yes?

Reply:
@AllaMaevskaya what your real firmware does? Do you split the data of the BC if does not fit to current 8KB page or just open a page? If you always create a new page w/o splitting, then your carryover method should always return 0.
If you split, then it is better to implement this properly.

Reply:
Data of one BC will never be split, just new page will be open and header+data (7 words max) will be written there.  What is wrong here?

Reply:
I can ask electronics people again, but before it was like this

Reply:
Hi @AllaMaevskaya 

If it is never split, which makes sense for compact payload per BC, then just ``return 0;``, this comparison ``data.size() > maxSize ? 0 : data.size();`` is meaningless.

Pinging to @mslupeck and @mbroz84 : I guess the same carryover method should be also implemented for FV0 and FDD (in case their per-BC data also cannot exceed 8K).

Reply:
Dear All,
in case of ++8kB of payload per HBF, before exceeding 8kB, next event will be moved to next page as one piece(header+data). This logic is implemented into FEE FW.

Reply:
thanks @afurs , then the carryover method should simply always return 0.

Reply:
@shahor02 Thanks for the notification - @AllaMaevskaya already made us aware of it

Reply:
Hi @AllaMaevskaya, could you please apply the fix I mentioned above? @afurs : what about the uninitialized variable error?

Reply:
Hello @shahor02 , 
Is it possible to add changes within this PR?

There are only one line should be replaced:
https://github.com/AliceO2Group/AliceO2/blob/17391fc5cbcb0ca54020e0f0c714a20ac95e4e00/Detectors/FIT/raw/include/FITRaw/RawReaderBase.h#L49
to
`DataBlockType dataBlock{};`

Reply:
sorry, too late, I've merged just before you wrote. Please open a new PR, I'll merge it after the clang-format test.